### PR TITLE
feat: Swap modal tab system

### PR DESCRIPTION
**Pull request by crnbarr93**
## What is the purpose of the change:
These changes add a simple tab system for ths swap modal in order to allow options for placing limit orders. The tabs include three options:
1. Buy 
2. Sell
3. Swap

With the first and second options being specifically for limit orders and the 3rd being the current swap order implementation. These tabs currently don't alter the swap modal in any way but are set up to allow a parent component to record and update state depending on the tab selected.

### Linear Task

TODO

## Brief Changelog
- Added `SwapToolTabs` component
- Added `activeTab` state to `SwapTool`

## Testing and Verifying
No unit-tests or integration tests were written for this component

### CODE REVIEW

**Review by fabryscript — CHANGES_REQUESTED** [review]
I think there is no need to re-invent the wheel for the tabs system

Here is a good alternative with helper styles: https://headlessui.com/react/tabs

File: packages/web/components/swap-tool/swap-tool-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  // Renders each tab as a button, providing a highlight on the current `activeTab`
+  const Tabs = useMemo(() => {
+    return tabs.map((tab) => {
+      const isActive = activeTab === tab.value;
+      return (
+        <button
+          key={`swap-tab-${tab.value}`}
+          onClick={() => setTab(tab.value)}
+          className={classNames(
+            "text-subtitle6 justify-center rounded-[16px] border-2 py-[12px] text-center font-subtitle1 transition-all duration-75",
```
Comment (fabryscript) [inline packages/web/components/swap-tool/swap-tool-tabs.tsx:50]: you can use `py-3` and `rounded-2xl`

File: packages/web/components/swap-tool/swap-tool-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          )}
+        >
+          {tab.label}
+        </button>
+      );
+    });
+  }, [setTab, activeTab]);
+
+  return (
+    <div className="grid grid-cols-3 rounded-[16px] bg-osmoverse-900 transition-opacity">
```
Comment (fabryscript) [inline packages/web/components/swap-tool/swap-tool-tabs.tsx:64]: you can use `rounded-2xl`

**Review by fabryscript — CHANGES_REQUESTED** [review]

**Review by jonator — COMMENTED** [review]
It seems the designs are very much still in flux that are quite different from this implementation.

Should we hold of or close this until designs are finalized.

Was inspired by this:
https://osmosis-network.slack.com/archives/C06PW1BHKC3/p1717172538599349

**Comment by crnbarr93** [comment]
> It seems the designs are very much still in flux that are quite different from this implementation.
> 
> Should we hold of or close this until designs are finalized.
> 
> Was inspired by this: https://osmosis-network.slack.com/archives/C06PW1BHKC3/p1717172538599349

Happy to hold off on this PR until designs are finalized.

**Pull request closed without merging**